### ISSUE TITLE
Optionally show file name as permanode title line

### ISSUE THREAD

**Issue by glowingwire** [camlistore/camlistore#746]
I was unable to see the filename when I dragged it in, and looked at image, permanode, and blob.

I keep a lot of metadata in my filenames.
I uploaded a file with the title veep - s01e02.mp4  (wasn't actually a veep episode), searched for veep in camlistore web interface and the file didn't show up.
I'm guessing that the fuse system keeps this data, and I want to play with that later.

**Comment by willnorris** [comment]
in the blob view of the permanode, click on the camliContent blobref, which should take you to the file blob for the image.  That has the fileName.  It's also visible in the UI when you hover over the image for a few seconds.

**Comment by mpl** [comment]
The filename is indeed definitely stored and indexed.
And in addition to what @willnorris said, you can also use the "filename" search predicate to look for your file (https://camlistore.org/doc/search-ui), although it has to be an exact match.

**Comment by glowingwire** [comment]
When I uploaded a text file named a1.txt it kept the name.
When I uploaded the mp4 file, it did not keep the name.
It went to the camliContent blobref and it did not have anything to do with the original filename.

Ok.. Well I did get to it... under Indexer metadata then camliContent... which took me back to the viewing mode, then when I click on blob it shows the filename on blob content.

The filename still doesn't show up in the main screen for the mp4 file, but it does for the text file.

**Comment by mpl** [comment]
Seeing as I don't really understand what your problem is, could you please restate exactly what is the behavior you would expect and you're not provided with? Or maybe what feature it is that you're wishing for and that is not there?

I indeed see that when I hover the cursor over a permanode for an .mp4 file, the filename does not show up as it does for another kind of file. Is that what this bug is about?

**Comment by glowingwire** [comment]
It might be a feature...
I would expect that the filename for an mp4 file shows up like the filename for the .txt file shows up.

Also, I would expect the search to do partial matches on filenames  (I would expect saying filename: would limit the results to filenames rather than be the only way to search for filenames)

**Comment by glowingwire** [comment]
![screen shot 2016-05-12 at 5 28 12 am](https://cloud.githubusercontent.com/assets/1845049/15216286/6974b016-180b-11e6-9d86-df36258333c4.png)
Here is a screenshot..

**Comment by glowingwire** [comment]
I'm also surprised that there is no preview for text documents... and there is for videos

**Comment by mpl** [comment]
Yes, I had forgotten that we get a file name below for txt files, I understand now.

I believe the initial intent (probably @aboodman 's) was that the estate space taken by the title line (in which the file name shows up) is wasted on e.g. on a picture, where you already get a snapshot. So the choice was to "reclaim" that space to get you a slightly higher snapshot rather than giving you a title.

Same goes for videos, you should get a snapshot for them, which makes the need for a title less useful. It just happens that creating a snapshot is apparently not supported for the video you added. I have a CL to make that work for all of them video types with ffmpeg (https://camlistore-review.googlesource.com/5116) but it got forgotten.

So yeah it's just a stance on how to represent each type.

To sum up:
1) content types (files) that do not have any special representation (txt files, or pretty much anything that is not an image or a video), do have a title line which usually defaults to the file name.
2) because video snapshots are not always working (which in itself is a bug), you get a representation for them that is pretty poor, and makes one feel that a title line could be useful there.

Is the situation clearer now?

**Comment by mpl** [comment]
And to answer your question about text documents: feel free to file a specific issue about it and/or contribute a patch for it. :-)

**Comment by glowingwire** [comment]
I think it is, using the filename as a title would be great for all types of media.
and that it should be easily searchable.
Should I change the summary for this to be show filename as title?
and make two new requests, 
one for searching filename
and one for viewing text files?

I put a lot of effort into selecting filenames, and snapshots from videos are rarely useful for me..  I suppose it would be nice to make it an option to show it or not?
I would expect the filename to be shown for everything or for nothing.

I'll create a feature request for the

**Comment by mpl** [comment]
1) Yes, for changing the title of this issue to something like "Always show file name as permanode title line" (although I can't guarantee we'll agree with that feature)

2) Yes, please create a new issue for text file previews

3) I didn't understand what you were saying about filename searching, but please do explain it in a different issue too.

**Comment by glowingwire** [comment]
I can't edit this.. So making new issue

**Comment by glowingwire** [comment]
Note, I had to go 4 steps to even see the filename, it wasn't just going to blob or permanode.

**Comment by mpl** [comment]
@glowingwire said on issue #769 : 

"I discovered that a text file shows filename, but an mp4 file doesn't. This is regarded as a feature. Since I put so much effort into filenames, and folder structure, seeing these would be great help. I propose to always show the filename as object title, because I will never not want the filename, but other people may want it off, so option is requested.

It is hard to search for filenames so seeing them is important."
